### PR TITLE
Add max length option to BranchNamer

### PR DESCRIPTION
**Pull request by TomNaessens**
This PR adds a max length option to the BranchNamer and addresses https://github.com/dependabot/dependabot-core/issues/3107. Also relevant for the "... and may contain a maximum of 128 characters" part of https://github.com/dependabot/dependabot-core/issues/396.

This is a backwards compatible change and requires no configuration: when no option is passed, no truncating is performed on the branch name.

As this is my first contribution here, let me know if I missed some conventions, or if there are additional changes expected!

### CODE REVIEW

File: common/lib/dependabot/pull_request_creator/branch_namer.rb [hunk 1]
```diff
@@ -37,7 +38,14 @@ def new_branch_name
           end
 
         # Some users need branch names without slashes
-        sanitize_ref(File.join(prefixes, @name).gsub("/", separator))
+        sanitized_name = sanitize_ref(File.join(prefixes, @name).gsub("/", separator))
+
+        # Shorten the ref in case users refs have length limits
+        if @max_length
+          sanitized_name[0, @max_length]
```
Comment (jurre) [inline common/lib/dependabot/pull_request_creator/branch_namer.rb:45]: I'm worried that we'd end up with duplicate branch names in this scenario, as we'd end up truncating the bit that makes them unique (the version), and I think that could cause issues

Comment (nudded) [reply common/lib/dependabot/pull_request_creator/branch_namer.rb:45]: That's a valid concern.

Would you be okay with taking a hash of the full-length branch name and taking the first `max_length` characters from that if `max_length` is specified? (At some point we'd have to choose the lesser of 2 evils, we cannot deal with going over the limit, however then you do lose pretty branch names)

Comment (jurre) [reply common/lib/dependabot/pull_request_creator/branch_namer.rb:45]: Yeah a hash could also work, it's a tad safer I guess (if for some reason the combination of dependency names changes, it's still unique)

**Comment by jeffwidman** [comment]
Approving the CI runs for first-time contributor as the PR seems headed in a generally useful direction.

**Comment by TomNaessens** [comment]
Sorry, took a while to get the feedback processed. Let me know what you think of the changes!

**Comment by SimonSomlai** [comment]
@TomNaessens , thanks for opening this, we've also ran into a similar problem. Seems like there are just some linter issues left (most of which can be autocorrected);

Screenshot 2022-09-13 at 10 11 37

@jeffwidman / @jurre, does this need more work, or do the changes look good? :)

**Comment by TomNaessens** [comment]
> @TomNaessens , thanks for opening this, we've also ran into a similar problem. Seems like there are just some linter issues left (most of which can be autocorrected);

Thanks! Should be fixed now 🙇‍♂️

File: common/lib/dependabot/pull_request_creator/branch_namer.rb (hunk 1 above)
Comment (jeffwidman) [reply common/lib/dependabot/pull_request_creator/branch_namer.rb]: Very clean solution @TomNaessens 👍

**Review by jeffwidman — APPROVED** [review]

**Comment by jeffwidman** [comment]
Rebased, once CI passes I'll deploy to production and assuming that looks good will merge. It's getting late here, so may not finish that til tomorrow though.

**Pull request merged**